docs/listeners/jsonapi.rst

```diff
@@ -0,0 +1,723 @@
+code ``400`` with empty response body.
+
+An valid JSON API document structure for deleting a Country
+with ``id`` 10 could look similar to:
+
+.. code-block:: json
+
+  {
+    "data": {
+      "type": "countries",
+      "id": "10"
+      }
+    }
+  }
+
+Associated data
+---------------
+
+The listener will detect associated data as produced by
+``contain`` and will automatically render those associations
+into the JSON API response as specified by the specification.
+
+Let's take the following example code for the ``view`` action of
+a Country model with a ``belongsTo`` association to Currencies
+and a ``hasMany`` relationship with Cultures:
+
+.. code-block:: php
+
+  public function view()
+  {
+    $this->Crud->on('beforeFind', function (Event $event) {
+      $event->subject()->query->contain([
+        'Currencies',
+        'Cultures',
+      ]);
+    });
+
+    return $this->Crud->execute();
+  }
+
+Assuming a succesful find the listener would produce the 
+following JSON API response including all associated data:
+
+.. code-block:: json
+
+  {
+    "data": {
+      "type": "countries",
+      "id": "2",
+      "attributes": {
+        "code": "BE",
+        "name": "Belgium"
+      },
+      "relationships": {
+        "currency": {
+          "data": {
+            "type": "currencies",
+            "id": "1"
+          },
+          "links": {
+            "self": "/currencies/1"
+          }
+        },
+        "cultures": {
+          "data": [
+            {
+              "type": "cultures",
+              "id": "2"
+            },
+            {
+              "type": "cultures",
+              "id": "3"
+            }
+          ],
+          "links": {
+            "self": "/cultures?country_id=2"
+          }
+        }
+      },
+      "links": {
+        "self": "/countries/2"
+      }
+    },
+    "included": [
+      {
+        "type": "currencies",
+        "id": "1",
+        "attributes": {
+          "code": "EUR",
+          "name": "Euro"
+        },
+        "links": {
+          "self": "/currencies/1"
+        }
+      },
+      {
+        "type": "cultures",
+        "id": "2",
+        "attributes": {
+          "code": "nl-BE",
+          "name": "Dutch (Belgium)"
+        },
+        "links": {
+          "self": "/cultures/2"
+        }
+      },
+      {
+        "type": "cultures",
+        "id": "3",
+        "attributes": {
+          "code": "fr-BE",
+          "name": "French (Belgium)"
+        },
+        "links": {
+          "self": "/cultures/3"
+        }
+      }
+    ]
+  }
+
+.. note::
+
+Please note that only support for ``belongsTo`` and ``hasMany``
+relationships has been implemented.
+
+
+Configuration
+-------------
+
+The output produced by the listener is highly configurable using the Crud configuration
+options described in this section. Configure the options on the fly in your controller
+actions or enable them for all actions in your controller by adding them to the
```

Comment:
'by adding them to your controllers' I would think here, just an additional 'the'